src/llmq/quorums.cpp

```diff
@@ -166,6 +166,17 @@ void CQuorumManager::UpdatedBlockTip(const CBlockIndex* pindexNew, bool fInitial
     for (auto& p : Params().GetConsensus().llmqs) {
         EnsureQuorumConnections(p.first, pindexNew);
     }
+
+    // Remove outdated quorums from the cache
+    LOCK(quorumsCacheCs);
+    auto it = quorumsCache.begin();
+    while (it != quorumsCache.end()) {
+        if ((pindexNew->nHeight - it->second->pindexQuorum->nHeight) > CLLMQUtils::GetQuorumCacheDepth(it->first.first)) {
```

Comment:
This won't work if new quorums would fail to form for some reason. Should probably switch `quorumsCache` from `map` to `unordered_lru_cache` to achieve "limited cache" effect (i.e. make it like `scanQuorumsCache`, but double the size). (sidenote: should bump `scanQuorumsCache` size from 32 to 32+24 because of `llmq100_67`, my bad, missed it when introduced `llmq100_67` 🙈 )

Reply:
Why do you mean it won't work then? If it fails to form the quorum will just not end up in `quorumsCache` anyway or do i miss something?

Reply:
I mean if this happens you'll start to drop quorums from the cache because the chain would still grow. As a result, node performance is going to degrade with each quorum pushed out of the cache.

Reply:
Ahh, right! Good point 👍